packages/react-searchbox/src/components/SearchComponent.js

```diff
@@ -189,6 +194,32 @@ class SearchComponent extends React.Component {
       stopwords,
       libAlias: LIBRARY_ALIAS.REACT_SEARCHBOX
     });
+    if (context.initialState) {
+      const initialState = context.initialState[id];
+      if (initialState) {
+        this.componentInstance.aggregationData.data =
+          initialState.aggregationData.data;
+        this.componentInstance.aggregationData.setRaw(
+          initialState.aggregationData.rawData
+        );
+        this.componentInstance.categoryValue = initialState.categoryValue;
+        this.componentInstance._queryId = initialState.queryId;
+        this.componentInstance.value = initialState.value;
+        this.componentInstance._lastRequestTimeCustomQuery =
+          initialState._lastRequestTimeCustomQuery;
+        this.componentInstance._lastRequestTimeDefaultQuery =
+          initialState._lastRequestTimeDefaultQuery;
+        this.componentInstance.results.data = initialState.results.data;
+        this.componentInstance.results.raw = initialState.results.raw;
+        this.componentInstance.results.setRaw(initialState.results.rawData);
+        if (Array.isArray(JSON.parse(initialState._query))) {
+          this.componentInstance._query = JSON.parse(initialState._query)[0];
```

Comment:
@mohdashraf010897 Why are you using first element of the query? Can you explain the logic here?

Reply:
because on the initial render, the `_query` would be compared to  `componentQuery` when making defaultQuery/ customQuery call

And the componentQuery is an object, rather than an array.

Also check the image below, line 1321 would tell that the first element in the finalQuery array is the componentQuery. 
<img width="896" alt="image" src="https://user-images.githubusercontent.com/57627350/165624091-4266c82c-676b-4823-b879-7c5391523117.png">